tidb-server/main.go

```diff
@@ -203,6 +204,8 @@ func main() {
 	// For regression of issue like https://github.com/pingcap/tidb/issues/28190
 	terror.RegisterFinish()
 
+	charset.RegisterFinish()
```

Comment:
it still can not avoid data race in tests? the tests never run here

Reply:
Yes, tests couldn't run in parallel.

Reply:
then, this pr cannot close the issue